documentation/network/ble/ble.md

```diff
@@ -15,7 +15,7 @@ This document describes the Moddable SDK Bluetooth Low Energy (BLE) modules. Bot
 * [Using BLE](#usingble)
 * [BLE Client](#bleclient)
 	* [Class BLEClient](#classbleclient)
-	* [Class Device](#classdevice)
+	* [Class Client](#classclient)
```

Comment:
Thank you for the contribution! It will be very helpful to everyone to have TypeScript support for BLE.

One question: what is the reasoning behind changing the name of the `Device` class to `Client`? The BLE devices described by this class can be either BLE Server or BLE Client devices. And in this context—devices being discovered by a BLE Client—they will almost always be BLE Server devices. (More broadly, the devices described by this class can be either BLE Peripherals or BLE Centrals and either GATT Servers or GATT Clients.)

I think I would prefer a different name here. If there's a conflict on `Device`, perhaps just `BLEDevice`?

Reply:
Hi, exactly it is because I see the `Client` defined in GATT module:
https://github.com/Moddable-OpenSource/moddable/blob/public/modules/network/ble/gatt.js#L23
This client definition seems consistent with the object device intialization across both `BLEServer` and `BLEClient`:
https://github.com/Moddable-OpenSource/moddable/blob/public/modules/network/ble/bleserver.js#L141
https://github.com/Moddable-OpenSource/moddable/blob/public/modules/network/ble/bleclient.js#L110

I believe the `Client` here would be more matching the reality where the client class is defined.

Also, I cannot find where is the class called `Device` exactly defined in this code base, I thought that the writing of Device seems a typo regarding this definition missing in code.

But if there is exactly the `Device` defined as an abstract class which I didn't recognize correctly, please let me know.

Reply:
Also if you prefer `BLEDevice` than `Client`, I also hope this class naming could be initially employed in the real implementation codes, so that it can become a reference to support the type definition in TypeScript.

Reply:
Ah, I understand now; thank you for the clarification. 

I do think that the `Client` class that you reference in `gatt.js` is misnamed. It has not previously been an issue (and I had not noticed it) because that class is generally only instantiated by internal Moddable SDK code and then presented to scripts as an instance of the abstract `Device` class.

You can see that in action in examples that use the `onConnected` callback. For example, while `bleclient.js` references `Client`:

```js
const client = new Client({ address, addressType, connection:params.connection, ble });
this.onConnected(client);
```
the `onConnected` callbacks in script generally look like (this example from `uartclient.js`:
```js
onConnected(device) {
	device.discoverPrimaryService(SERVICE_UUID);
}
```

At any rate, I now agree that (if anything) this is a flaw in `bleclient.js` and `gatt.js`, not in your proposed Typing. So, I'm fine with taking this as-is, and then I'll consider whether there's something to be fixed on our side.